src/RealtimeClient.ts

```diff
@@ -150,55 +164,27 @@ export default class RealtimeClient {
   constructor(endPoint: string, options?: RealtimeClientOptions) {
     this.endPoint = `${endPoint}/${TRANSPORTS.websocket}`
     this.httpEndpoint = httpEndpointURL(endPoint)
-    if (options?.transport) {
-      this.transport = options.transport
-    } else {
-      this.transport = null
-    }
-    if (options?.params) this.params = options.params
-    if (options?.timeout) this.timeout = options.timeout
-    if (options?.logger) this.logger = options.logger
-    if (options?.logLevel || options?.log_level) {
-      this.logLevel = options.logLevel || options.log_level
-      this.params = { ...this.params, log_level: this.logLevel as string }
+    
+    // Validate required options
+    if (!options?.params?.apikey) {
+      throw new Error('API key is required to connect to Realtime')
     }
-
-    if (options?.heartbeatIntervalMs)
-      this.heartbeatIntervalMs = options.heartbeatIntervalMs
-
-    const accessTokenValue = options?.params?.apikey
-    if (accessTokenValue) {
-      this.accessTokenValue = accessTokenValue
-      this.apiKey = accessTokenValue
-    }
-
-    this.reconnectAfterMs = options?.reconnectAfterMs
-      ? options.reconnectAfterMs
-      : (tries: number) => {
-          return [1000, 2000, 5000, 10000][tries - 1] || 10000
+    this.apiKey = options.params.apikey
+    
+    // Initialize all options with defaults
+    this._initializeOptions(options)
+    
+    // Set up reconnection timer
+    this.reconnectTimer = new Timer(() => {
+      setTimeout(() => {
+        if (!this.isConnected()) {
+          this.connect()
         }
-    this.encode = options?.encode
-      ? options.encode
-      : (payload: JSON, callback: Function) => {
-          return callback(JSON.stringify(payload))
-        }
-    this.decode = options?.decode
-      ? options.decode
-      : this.serializer.decode.bind(this.serializer)
-    this.reconnectTimer = new Timer(async () => {
-      this.disconnect()
-      this.connect()
+      }, CONNECTION_TIMEOUTS.RECONNECT_DELAY)
```

Comment:
why delaying for calling the `connect`?

Reply:
prevents timing issues of calling this at the wrong time (e.g. while setAuth is being set)